### ISSUE TITLE
Renouncing Ownership

### ISSUE THREAD

**Issue by taitruong** [larry0x/cw-plus-plus#18]
This action is quite critical and executes it without confirmation. If owner e.g. accidentally does this, contract will be immutable once and forever.

I believe existing `TransferOwnership` action can be adjusted by having `new_owner: Option<String>` instead of `new_owner: String`. Then in `AcceptOwnership` we just need to check whether sender is owner and then set owner `None`. This way `RenounceOwnership` would be obsolete.

@larry0x, wdyt?

**Comment by Unique-Divine** [comment]
Chiming in as another team that uses `cw-ownable` a lot: Using `new_owner: Option<String>` seems to make this more error prune for the person using it to have `new_owner == None` imply that you want to `RenounceOwnership`. For example, if you send the message with the CLI or the cosmjs client and forget the `new_owner` argument, defaulting to renounce would be a bit dangerous.

**Comment by taitruong** [comment]
@CyberHoward pls have a look at this. Imo we can simplify this and make it even more secure.

**Comment by cyberhoward** [comment]
I don't think this is worth changing the API for. Having the ability to do destructive actions will always be present when managing smart-contracts. It's up to the person(s) that manage these smart-contracts to ensure that only the correct calls are performed. Be it through the use of a multi-sig or other tooling.
